returnn/datasets/lm.py

```diff
@@ -178,26 +178,27 @@ def __init__(self,
     if word_end_symbol and not word_based:  # Character-based modeling and word_end_symbol is specified.
       self.orth_replace_map[" "] = [word_end_symbol]  # Replace all spaces by word_end_symbol.
 
-    num_labels = len(self.labels["data"])
```

Comment:
I don't see why you changed this. This should always be correct. If not, then `labels` is wrong.

Reply:
there is a problem if two or more tokens get mapped to the same output e.g.
```
<s> 0
<\s> 0
<UNK> 1
hello 2
world 3
```
in this case `num_labels` would be 5 but we only want 4 outputs.

Reply:
If `labels` is correct, then `num_labels = 4`. If `num_labels != 4`, then `labels` is incorrect.

Reply:
What you listed is not what `labels` is. `labels` is just a list, with the condition that `labels[label_idx] == label`.

Reply:
> then maybe the code that generates the `labels` from the `orth_symbols_map` is wrong?

Yes this looks wrong. At least it sounds as if `labels` are wrong here. Because we always must have `len(labels) == num_labels`.

Reply:
Yes, `labels` is wrong here, see: https://github.com/rwth-i6/returnn/issues/1245#issuecomment-1373451235